src/js/stores/search.js

```diff
@@ -271,38 +272,46 @@ define(function (require, exports, module) {
          */
         _getOptions: function (type) {
             var searchTypeInfo = this._registeredSearchTypes[type];
+            
             if (searchTypeInfo) {
                 var items = searchTypeInfo.getOptions();
-                
-                // Get shortest unique paths
-                var ancestors = collection.pluck(items, "pathInfo"),
-                    shortenedPaths = searchTypeInfo.shortenPaths ?
-                        pathUtil.getShortestUniquePaths(ancestors).toJS() : ancestors.toJS();
-
-                var itemMap = items.map(function (item, index) {
-                    var newPathInfo = shortenedPaths[index] || "",
-                        name = item.name,
-                        style;
-                    // Don't show the path info if it is just the same as the item name 
-                    if (name === newPathInfo) {
-                        newPathInfo = "";
-                    }
-
-                    if (name === "") {
-                        name = "Untitled";
-                        style = { "fontStyle": "italic" };
-                    }
-
-                    return {
-                        id: type + "-" + item.id.toString(),
-                        title: name,
-                        pathInfo: newPathInfo,
-                        svgType: item.iconID,
-                        category: item.category,
-                        style: style,
-                        type: "item"
-                    };
-                }.bind(this));
+                var itemMap;
+
+                if (items === null) {
+                    itemMap = [];
+                } else {
+                    // Get shortest unique paths
+                    var ancestors = collection.pluck(items, "pathInfo"),
+                        shortenedPaths = searchTypeInfo.shortenPaths ?
+                            pathUtil.getShortestUniquePaths(ancestors).toJS() : ancestors.toJS();
```

Comment:
There's no need to call `toJS` here, which is slow. Instead of doing an array lookup below, you can not convert these immutable objects and call `.get(index)`. (I know you didn't create this bug, but you're in there now...)